File: leet_2PTR_1750_min_len.cc
class Solution {
public:
    int minimumLength(string s) {
        int N = (int)s.length();
        int l = 0, r = N - 1;
        char c;
        while (l < r) {
            if (s[l] == s[r]) {
                c = s[l];
                while (l <= r && c == s[l]) {
                    ++l;
                }
                while (l <= r && c == s[r]) {
                    --r;
                }
            } else break;
        }
        return r + 1 - l;
    }
};

File: leet_2PTR_1750_min_len.py
class Solution:
    def minimumLength(self, s: str) -> int:
        N = len(s)
        l, r = 0, N - 1
        while l < r:
            if s[l] == s[r]:
                c = s[l]
                while l <= r and s[l] == c:
                    l += 1
                while l <= r and s[r] == c:
                    r -= 1
            else:break
        # print(len(s[l:r+1]), s[l:r+1])
        return r - l + 1
